src/site/content/en/notifications/use-push-notifications-to-engage-users/index.md

```diff
@@ -0,0 +1,162 @@
+---
+layout: post
+title: Use push notifications to engage and re-engage users
+authors:
+  - katejeffreys
+description: |
+  Use push notifications to engage users with targeted, timely updates.
+date: 2019-10-21
+codelabs:
+  - codelab-notifications-get-started
+  - codelab-notifications-service-worker
+---
+
+## Why use push notifications?
+
+Notifications present small chunks of information to a user. Web apps can use notifications to tell users about important, time-sensitive events, or actions the user needs to take. 
+
+The look-and-feel of notifications varies between platforms. For example: 
+
+<figure class="w-figure w-figure--center">
+  <img class="w-screenshot w-screenshot--filled" src="./predicaments-android.png" alt="">
+  <figcaption class="w-figcaption">A notification on an Android device.</figcaption>
+</figure>
+
+<figure class="w-figure w-figure--center">
+  <img class="w-screenshot w-screenshot--filled" src="./predicaments-macbook.png" alt="">
+  <figcaption class="w-figcaption">A notification on a MacBook.</figcaption>
+</figure>
+
+Traditionally, web browsers had to initiate the exchange of information between server and client by making a request. [Web push technology](https://developer.mozilla.org/en-US/docs/Web/API/Push_API), on the other hand, lets you configure your server to send notifications when it makes sense for your app. A push service creates unique URLs for each subscribed service worker. Sending messages to a service worker's URL raises events on that service worker, prompting it to display a notification.      
+
+Push notifications can help users to get the most out of your app by prompting them to re-open it and use it based on the latest information. 
+
+## Creating and sending notifications
+
+Create notifications using the [Notifications API](https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API). A `Notification` object has a `title` string and an `options` object. For example:
+
+```js
+let title = 'Hi!';
+let options = { 
+  body: 'Very Important Message',
+  /* other options can go here */
+};
+let notification = new Notification(title, options);
+```
+
+The `title` is displayed in bold when the notification is active, while the `body` contains additional text.
+
+### Get permission to send notifications
+
+To display a notification, your app must get permission from the user to do so: 
+
+```js
+Notification.requestPermission();
+```
+
+{% Aside %}
+
+The web platform encourages developers to request notifications permission when users understand what is being asked and why. 
+
+Bring up the permission prompt **in response to a user gesture** and **when the value of notifications is obvious**. Avoid bringing up the permission prompt immediately when the user lands on the page.
```

Comment:
Might be worth giving an example of 'in response to a user gesture'.

Maybe something like 'For example, describe briefly what your notifications have to offer, and provide the user with Sign Up button, rather than simply displaying the Notifications permission dialog.'

On a tangent: do we have any data that shows this is actually best practice?>